### PR TITLE
Updating the MSAL section

### PR DESCRIPTION
**Pull request by hhawkins**
This update to the MSAL section is to ensure the wording and organization is optimal to increase clarity and reduce some common implementation issues we've been noticing. This also fixes some of the broken links.

### CODE REVIEW

**Comment by PRMerger18** [comment]
@hhawkins : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by ktoliver** [comment]
#label:"aq-pr-triaged"

File: memdocs/intune/developer/app-sdk-ios.md [hunk 1]
```diff
@@ -170,30 +170,33 @@ In `- startProvidingItemAtURL:completionHandler:` check if you should encrypt fi
...
 
-Typically, MSAL requires apps to register with Azure Active Directory (AAD) and create a unique client ID and redirect URI, to guarantee the security of the tokens granted to the app. If your app already uses MSAL to authenticate users, the app must use its existing registration values and override the Intune App SDK default values. This ensures that users are not prompted for authentication twice (once by the Intune App SDK and once by the app).
+Typically, MSAL requires apps to register with Azure Active Directory (AAD) and create a unique client ID and redirect URI, to guarantee the security of the tokens granted to the app. If your app already uses MSAL to authenticate users, the app must use its existing registration values and override the Intune App SDK default values. This ensures that users are not prompted for authentication twice (once by the Intune App SDK and once by the app).
```
Comment (vigavadi) [inline memdocs/intune/developer/app-sdk-ios.md:177]: Is there an extra space at the end?

**Comment by ktoliver** [comment]
@Erikre for review

Thanks.

**Comment by traya1** [comment]
Hi @Erikre,

Could you review the PR and let me know when you approve of the proposed changes? 

Thanks!

**Pull request closed without merging**